### DISCUSSION TITLE
Generate AST, modify it and write back to markdown

### DISCUSSION THREAD

**Discussion by sewid** [thephpleague/commonmark#800] [Q&A]
Hi there,

I try to generate the AST from a markdown file, modify the AST (translate it) and write it back to markdown. Is this somehow supported or do I have to write my own Markdown Renderer, when I want to write the AST back to markdown?

Another idea might by to generate a XML file from markdown, modify this and write back the XML to markdown (how?).

Thanks,
Sebastian

**Comment by colinodell** [comment]
This library doesn't yet have that functionality, but I'm hoping to implement it for #419 in the next minor release (if possible).  It would basically allow any AST to be rendered to Markdown (we currently support rendering to HTML and XML).

**Comment by sewid** [comment]
That sounds great, thanks for the update!
